pkg/operator/target_config_reconciler.go

```diff
@@ -115,26 +116,35 @@ func (c TargetConfigReconciler) sync() error {
 	}
 
 	forceDeployment := false
-	replicas := int32(1)
 	_, forceDeployment, err = c.manageConfigMap(descheduler)
 	if err != nil {
 		// if we returned an error from the configmap AND want to force a deployment
 		// it means we want to scale the deployment to 0
 		if forceDeployment {
-			replicas = 0
-			deployment, _, err := c.manageDeployment(descheduler, forceDeployment, &replicas)
+			klog.ErrorS(err, "Error managing targetConfig")
+			_, err := c.kubeClient.AppsV1().Deployments(operatorclient.OperatorNamespace).UpdateScale(
+				c.ctx,
+				descheduler.Name,
+				&autoscalingv1.Scale{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      descheduler.Name,
+						Namespace: operatorclient.OperatorNamespace,
+					},
+					Spec: autoscalingv1.ScaleSpec{
+						Replicas: 0,
+					},
+				},
+				metav1.UpdateOptions{})
 			if err != nil {
-				klog.ErrorS(err, "Unable to scale descheduler operand deployment")
+				klog.ErrorS(err, "Error scaling descheduler operand deployment")
 			}
 			_, _, err = v1helpers.UpdateStatus(c.deschedulerClient,
 				v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
 					Type:   "TargetConfigControllerDegraded",
 					Status: operatorv1.ConditionTrue,
-				}),
-				func(status *operatorv1.OperatorStatus) error {
-					resourcemerge.SetDeploymentGeneration(&status.Generations, deployment)
-					return nil
-				})
+				}))
+			// return nil so the controller doesn't keep logging the error and the queue key is forgotten
```

Comment:
The key needs to be forgotten only when both the operator is in Degraded state and the descheduler deployment is scaled to 0. Otherwise, the descheduler might keep running or a user does not get notified. Unless there's another event that periodically triggers the reconciling loop which is independent of the scheduler config.

Reply:
Good point, updated this to just return `err` (if `nil`, great, if not then it will report+retry)